lib/scholar/metrics.ex

```diff
@@ -380,44 +367,11 @@ defmodule Scholar.Metrics do
   defn f1_score(y_true, y_pred, opts \\ []) do
     opts = keyword!(opts, [:num_classes, :average])
 
-    assert_rank(y_true, 1)
-    assert_same_shape(y_pred, y_true)
-
-    num_classes =
-      transform(opts[:num_classes], fn num_classes ->
-        num_classes || raise ArgumentError, "missing option :num_classes"
-      end)
-
-    transform(opts[:average], fn average ->
-      if Elixir.Kernel.==(average, :micro) do
```

Comment:
Actually, now we can use `case` directly in defn, so we can use it here :)

Reply:
Oh, ok